File: server/php/src/Application/Domain/Payments/PaymentDetails/PaymentProduct3203SpecificOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class PaymentProduct3203SpecificOutput
{
    public ?AddressPersonal $billingAddress = null;
    public ?AddressPersonal $shippingAddress = null;

    public function toArray(): array
    {
        return [
            'billingAddress' => ($this->billingAddress ?? new AddressPersonal())->toArray(),
            'shippingAddress' => ($this->shippingAddress ?? new AddressPersonal())->toArray(),
        ];
    }
}

File: server/python/app/application/domain/payments/payment_details/payment_product_3203_specific_output.py
from dataclasses import dataclass
from typing import Optional

from app.application.domain.payments.payment_details.address_personal import AddressPersonal

@dataclass
class PaymentProduct3203SpecificOutput:
    billing_address: Optional[AddressPersonal] = None
    shipping_address: Optional[AddressPersonal] = None